ui/app/ducks/swaps/swaps.js

```diff
@@ -563,9 +571,12 @@ export const fetchQuotesAndSetQuoteState = (
 
 export const signAndSendTransactions = (history, metaMetricsEvent) => {
   return async (dispatch, getState) => {
+    const state = getState();
+    const chainId = getCurrentChainId(state);
```

Comment:
[Q] would it be worth it to extract fetchSwapsFeatureLiveness into its own dispatch action / thunk so that anywhere we call it in the UI code it can be responsible for getting chainid itself? Seems like it'd cut down on repetition.

Reply:
I have made this change on another branch that I will PR once this one is merged: https://github.com/MetaMask/metamask-extension/pull/new/fetchswapsfeatureliveness-thunk